lib/resonate.ts

```diff
@@ -40,6 +40,8 @@ function isG<A extends any[], R>(g: unknown): g is G<A, R> {
 
 type ResonateOpts = {
   url: string;
+  // take db as an argument
+  db: IDBDatabase;
```

Comment:
I think we should pass in either a 
1. RemoteStore instance, or
2. Storage instance

And use this as the default store rather than pass in an IDBDatabase which is specific to an IndexedDbStorage. Perhaps we could do something like this in the constructor.

```ts
type ResonateOpts = {
  store: IPromiseStore
  storage: IStorage
}

let defaultStore: IPromiseStore;
if store() {
  defaultStore = store !== undefined ? store : new LocalPromiseStore(storage)
} else {
  defaultStore = url ? new RemotePromiseStore(url, logger) : new LocalPromiseStore()
}

this.addStore("default", defaultStore);
```

What do you think?

Reply:
This is a great idea. Implemented.